EIPS/eip-3074.md

```diff
@@ -56,39 +53,66 @@ The variable has the same scope as the program counter -- `authorized` persists
 
 ### `AUTH` (`0xf6`)
 
-A new opcode `AUTH` shall be created at `0xf6`. It shall take four stack element inputs and returns one stack element.
+A new opcode `AUTH` shall be created at `0xf6`. It shall take three stack element inputs (the last two describing a memory range), and it shall return one stack element.
 
 #### Input
 
+##### Stack
+
 | Stack      | Value        |
 | ---------- | ------------ |
-| `top - 0`  | `commit`     |
-| `top - 1`  | `yParity`    |
-| `top - 2`  | `r`          |
-| `top - 3`  | `s`          |
+| `top - 0`  | `authority`  |
+| `top - 1`  | `offset`     |
+| `top - 2`  | `length`     |
+
+##### Memory
+
+The final two stack arguments (`offset` and `length`) describe a range of memory. The format of the contents of that range is:
+
+ - `memory[offset    : offset+32 ]` - `yParity`
+ - `memory[offset+32 : offset+64 ]` - `r`
+ - `memory[offset+64 : offset+96 ]` - `s`
+ - `memory[offset+96 : offset+128]` - `commit`
 
 #### Output
 
+##### Stack
+
 | Stack      | Value        |
 | ---------- | -------------|
-| `top - 0`  | `authorized` |
+| `top - 0`  | `success`    |
+
+##### Memory
+
+Memory is not modified by this instruction.
 
 #### Behavior
 
+If `length` is greater than 128, the extra bytes are ignored. Bytes outside the range (in the event `length` is less than 128) are treated as if they had been zeroes.
```

Comment:
@SamWilsn this implies a limit of 128 bytes on the memory of this opcode. What if there is a signature which uses more than 128 bytes?

Reply:
This is also not very clear regarding the gas schedule. What is meant by "ignored"? I am implementing this, and this question has to do with the gas schedule. Is the memory length capped at 128? Or, if the length is larger, and the memory length (relative to offset) is smaller than that, should that memory be filled with zeros? Or is it the length indeed capped at 128?

Reply:
> @SamWilsn this implies a limit of 128 bytes on the memory of this opcode.

I hope that it implies that the opcode only reads the first 128 bytes.

> What if there is a signature which uses more than 128 bytes?

That would be defined in the EIP which introduces the new signature scheme.